docs/eBpfExtensions.md

```diff
@@ -270,6 +319,13 @@ the context descriptor (using the `ebpf_context_descriptor_t` type) to the eBPF
 client hosted by the Execution Context. The `result` output parameter holds the return value from the eBPF program
 post execution.
 
+In cases where the same eBPF program will be invoked sequentially with different context data (aka batch invocation),
+the caller can reduce the overhead of by using the batch invocation APIs. Prior to the first invocation, the batch
+begin API is called, which caches state used by the eBPf program and prevents the program from being unloaded. The
+caller is responsible for providing storage of large enough to store an instance of ebpf_execution_context_state_t and
+ensuring that it remain valid until calling the batch end API. Between the begin and end calls, the caller may call
+the batch invoke API multiple times to invoke the BPF program with minimal overhead.
```

Comment:
Should we state the conditions that would invalidate the cached state, e.g., switching threads or temporarily lowering from dispatch IRQL, etc.? Developers may intuit that you can't call the batch start routine once in DriverEntry and use it forever, but it would be good to call out the conditions.